swiftsimio/visualisation/projection.py

```diff
@@ -231,13 +232,34 @@ def project_pixel_grid(
     else:
         combined_mask = mask
 
-    common_arguments = dict(
-        x=(x[combined_mask] - x_min) / max_range,
-        y=(y[combined_mask] - y_min) / max_range,
-        m=m[combined_mask],
-        h=hsml[combined_mask] / max_range,
-        res=resolution,
-    )
+    xfinal = array((x[combined_mask] - x_min) / max_range)
+    yfinal = array((y[combined_mask] - y_min) / max_range)
+    mfinal = array(m[combined_mask])
+    hfinal = array(hsml[combined_mask] / max_range)
+    rescaled_box = array([box_x / max_range, box_y / max_range])
+
+    xall = xfinal.copy()
+    yall = yfinal.copy()
+    mall = mfinal.copy()
+    hall = hfinal.copy()
```

Comment:
Is there some way that we can work out if we're going to see the edges ahead of time..? Just thinking that we can skip this step.

Reply:
Not without knowing what the maximum smoothing length inside the visualisation region is. And that would require an extra calculation that is also not cheap.

According to the line profiler, this bit of code takes only ~5% of the total time in the function, so the gains are small.